src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +53,133 @@
 public class TranscriptServlet extends HttpServlet {
   private static final String TRANSCRIPT_XML_URL_TEMPLATE =
       "http://video.google.com/timedtext?lang=en&v=";
-  private static final String START_ATTRIBUTE = "start";
-  private static final String DURATION_ATTRIBUTE = "dur";
-  private static final String TEXT_TAG = "text";
-  private static final String TEST_VIDEO_ID = "3ymwOvzhwHs";
+  public static final String ATTR_START = "start";
+  public static final String ATTR_DURATION = "dur";
+  public static final String TAG_TEXT = "text";
+  public static final String PARAM_LECTURE = "lecture";
+  public static final String PARAM_LECTURE_ID = "id";
+  public static final String PARAM_VIDEO_ID = "video";
+
+  private DatastoreService datastore;
 
   @Override
   public void init() throws ServletException {
-    // TODO: Implement Transcript.
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  /**
+   * Initializes the servlet with {@code testDatastore} created during testing.
+   *
+   * <p>The unit tests need access to the datastore to check that doPost() puts
+   * the entities in datastore.
+   */
+  @VisibleForTesting
+  void init(DatastoreService testDatastore) {
+    datastore = testDatastore;
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: Pass the video ID from another servlet.
-    String transcriptXMLUrl = TRANSCRIPT_XML_URL_TEMPLATE + TEST_VIDEO_ID;
-    DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
-    final DocumentBuilder documentBuilder;
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Document document = getTranscriptXmlAsDocument(request).get();
+    putTranscriptLinesInDatastore(request, document);
+  }
+
+  /**
+   * Returns the transcript for a video as a document. Otherwise, returns Optional.empty()
+   * if there is a parsing error.
+   *
+   * @param request Indicates the video to extract the transcript from.
+   */
+  private Optional<Document> getTranscriptXmlAsDocument(HttpServletRequest request)
```

Comment:
For this method I would directly pass videoId as this is the only thing the method cares about. The method needs not know what the request is. In general it is better to pass directly the fields it needs. I would say if you needed multiple (4+) fields from request, then passing request would make sense.